File: 6.Inheritance/Lecture 27/Program5.java


class Parent{
	
	int x=10;
	String str ="name";

	Parent(){
		System.out.println("Parent Constructor");
		System.out.println(this);
	}

	void parentMethod(){

		System.out.println(x);
		System.out.println(str);

	}

}

class Child extends Parent{
	
	int x=20;
	String str="data";

	Child(){
	System.out.println("Child COnstructor");
	System.out.println(this);
	}	
	void childMethod(){
		System.out.println(x);
		System.out.println(str);
	}


public static void main(String[] args) {
	

	
	Child obj = new Child();
	System.out.println(obj);

	System.out.println(obj.x);
	System.out.println(obj.str);
	obj.parentMethod();



	System.out.println(obj.x);
	System.out.println(obj.str);
	obj.childMethod();

}
}

File: 8.mixin/33/Program5.dart

//mixin on

class Parent{
	
	void m1(){
		print("parent-m1");
	}
}

mixin Demo on Parent{
	
	void fun(){

		print("mixin-fun");
	}
}

class Normal  extends Parent with Demo{
	
}

void main(){
	
	Normal obj = new Normal();
	obj.fun();
	obj.fun();
}